bot/cogs/grades_cog/grades_cog.py

```diff
@@ -0,0 +1,265 @@
+import json
+import os
+import gzip
+import logging
+import collections
+import itertools
+
+import discord
+import pandas as pd
+import discord.ext.commands as commands
+from bot.messaging.events import Events
+import typing as t
+
+from bot.consts import Colors
+import bot.extensions as ext
+from spellchecker import SpellChecker # Here we will add a utility for correcting user input
+
+log = logging.getLogger(__name__)
+
+MIN_YEAR = 2014
+TAG_CHUNK_SIZE = 12*3
+CSV_FILES = [
+    '2014_Fall.csv',
+    '2015_Fall.csv',
+    '2015_Fall.csv',
+    '2016_Fall.csv',
+    '2017_Fall.csv',
+    '2018_Fall.csv',
+    '2019_Fall.csv',
+    '2020_Fall.csv',
+    '2014_Spring.csv',
+    '2015_Spring.csv',
+    '2015_Spring.csv',
+    '2016_Spring.csv',
+    '2017_Spring.csv',
+    '2018_Spring.csv',
+    '2019_Spring.csv',
+]
+
+ASSET_LOCATION = 'bot/cogs/grades_cog/assets/'
+
+class GradesCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.load_data()
+
+    def load_data(self):
+        self.grades_df = pd.DataFrame()
+        for file in CSV_FILES:
+            self.grades_df = self.grades_df.append(pd.read_csv(f'{ASSET_LOCATION}{file}'))
```

Comment:
So I think we have a few unnecessary additions. panda.read_csv can infer the .csv files as being zipped or not. Might be worthwhile keeping the csv's zipped. 
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html
_under compression_

Reply:
I did actually consider this, but due to the fact the new CSV files are only a few hundred kilobytes, you don't gain much from zipping them and keeping them human readable is valuable